src/classes/i18n.js

```diff
@@ -168,11 +171,48 @@ export default class i18n {
 
     async get_languages () {
 
-        this.debug('getting languages');
+        const service_version = this.config.get('service_version');
 
-        const response = await axios.get(this.languages_url);
+        let languages = {};
 
-        const languages = response.data;
+        try {
+            // force download
+            if (process.env.REFRESH_TRANSLATIONS) {
+                throw new Error('force-refresh');
+            }
+
+            // load languages and version from meta.json
+            languages = JSON.parse(
+                await fs_promises.readFile(this.metadata_file)
+            );
+
+            // redownload on version mismatch
+            if (languages.version !== service_version) {
+                throw new Error('version-mismatch');
+            }
+
+            /**
+             * Since we have a valid meta.json, we'll assume the previous
+             * download was successful
+             */
+            return;
+        }
+        catch (error) {
+            this.error('Verification error for', service_version, error);
+
+            // download from server on error or REFRESH_TRANSLATIONS
+            const response = await axios.get(this.languages_url);
+
+            languages = response.data;
+        }
+
+        languages.version = service_version;
+
+        // save meta.json
```

Comment:
If we move this after L226, then we ensure that all language files are downloaded properly, no?

Reply:
[`025fbe8` (#14)](https://github.com/anyTV/anytv-i18n/pull/14/commits/025fbe894f546f97b9b28c0d8bc9afb1cc7e7467)